### DISCUSSION TITLE
How to use 'has' on relational field?

### DISCUSSION THREAD

**Discussion by Skryvvara** [prisma/prisma#11429] [Q&A] [answered]
I don't know if I'm missing anything but it just can't get it to work.

this is my schema
```prisma
model Game {
  id                String     @id
  name              String
  is_free           Boolean
  short_description String
  header_image      String
  website           String
  developers        String[]
  publishers        String[]
  windows           Boolean
  mac               Boolean
  linux             Boolean
  metacriticScore   Int
  metacriticUrl     String
  categories        Category[]
  genres            Genre[]
  background        String
  storeUrl          String
}

model Category {
  id          String @id
  description String
  games       Game[]
}

model Genre {
  id          String @id
  description String
  games       Game[]
}
```
and this is my query and I though I could use has like this but apparently I'm wrong.
```ts
const games = await prisma.game.findMany({
      where: {
        AND: {
          name: { contains: name, mode: 'insensitive' },
          categories: {
            has: ['co-op']
          }
        }
      },
      take: limit + 1,
      cursor: cursor ? { id: cursor } : undefined,
      include: {
        categories: true,
        genres: true
      },
      orderBy: {
        id: 'asc'
      }
    });
```

How do I do that/is it even doable?

**Comment by austincrim** [comment, marked as answer]
Hey @xDarkyne 👋 ,

The `has` filter is only available on **lists of scalar** properties in Postgres, like `String` or `Int`. `categories` is not a scalar property, it is a relation property of type `Category`.

You might want to take a look at our [relation filters](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#relation-filters). 

You could instead write a query similar to this:
```ts
await prisma.game.findMany({
    where: {
      AND: {
        name: { contains: 'foo', mode: 'insensitive' },
        categories: {
          some: { // or every
            description: {
              equals: 'co-op'
            }
          }
        }
      }
    },
    ...
  })
```

Let me know if I can clarify anything!

**Reply by Skryvvara** [reply]
Yeah I got that by now. Sadly some and every don't cover my use-case. I solved this problem with the help of the prisma slack. Although it's working I think it's very annoying inconvenience when working with prisma. hasEvery would cover that use-case with minimal effort.

The solution for my problem is this:
```ts
const generateRelationFilter = (list: string[], relationName: string, column: string) => list.map((value) => ({ [relationName]: { some: { [column]: { contains: value, mode: 'insensitive' } } } }));

const games = await prisma.game.findMany({
    where: [
      AND: [
        { name: { contains: name, mode: 'insensitive' } },
        { AND: generateRelationFilter(categories, 'categories', 'description')  }
      ]
    ]
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
